pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/util/DependencyOrder.java

```diff
@@ -114,7 +119,11 @@ public static ResolvedComponentResult postOrderDependencies(
 
         // Prepare an ordered set of files in post-order.
         for (ResolvedComponentResult d : configurationPostOrderDependencies(configuration)) {
-            files.add(idToFileMap.get(d.getModuleVersion()));
```

Comment:
Why are we doing this conditionally now?
This is not the same behavior.

Reply:
To clarify ...
The `files.add` should be unconditional just as it was before.
There's no need for `else`.

Reply:
without the else, it would just add a `null` to files for missing artifacts. Is that helpful? I found it to be confusing while debugging an issue.

Reply:
I believe that we need to investigate deeper the reasons for this behavior then.
The files in resolved configuration whould be in the map.
If they are not, we have a different problem than just reporting the missing file.
I'd like to see us get to the root of this instead of just scratching the surface.

Reply:
If this is urgent, feel free to proceed because I won't be able to respond until tomorrow.
But please put a TODO to investigate this further if you do that.

Reply:
This statement "The files in resolved configuration should be in the map." is not true if an artifact cannot be resolved. This case is reproducible when dependencies are improperly defined in case of multi-variant builds, for example. There's an internal ticket on which you are added as a watcher. Please see that for more context.

I don't think that the current behavior is buggy, but 1) there's no use in adding a null to `files` which the if statement prevents and 2) warning the user about the artifact that was not resolved is actionable compared to just throwing a non-specific exception message.

This is not blocking anything, so it's not urgent IMO. Feel free to investigate more and approve when you are satisfied.

Reply:
Please commit the change.
That warning is useful for diagnostics.

The key here is "... when dependencies are improperly defined in case of multi-variant builds ..." which confirmed my suspicion that something deeper unexpected and **unrelated to PyGradle** is going on. I agree that this is very specific to an internal use-case and will have to be solved internally.

Reply:
Thanks @zvezdan